src/mode/mode_command.ts

```diff
@@ -1,14 +1,27 @@
+import * as _ from 'lodash';
 import {ModeName, Mode} from './mode';
 import {showCmdLine} from './../cmd_line/main';
 import * as vscode from 'vscode';
 
 export default class CommandMode extends Mode {
+    private keyHandler : { [key : string] : (position : vscode.Position) => void; } = {};
+    
     constructor() {
         super(ModeName.Command);
+        
+        this.keyHandler = {
+            ":" : (position) => { showCmdLine() },   
+            "h" : (position) => { vscode.commands.executeCommand("cursorLeft"); },        
+            "j" : (position) => { vscode.commands.executeCommand("cursorDown"); },        
+            "k" : (position) => { vscode.commands.executeCommand("cursorUp"); },      
+            "l" : (position) => { vscode.commands.executeCommand("cursorRight"); },
+            ">>" : (position) => { vscode.commands.executeCommand("editor.action.indentLines"); },
+            "<<" : (position) => { vscode.commands.executeCommand("editor.action.outdentLines"); }, 
+        };
     }
 
     ShouldBeActivated(key : string, currentMode : ModeName) : boolean {
-        return (key === 'esc');
+        return (key === 'esc' || key == 'ctrl+[');

```

Comment:
When we have mappings, we can make `ctrl+[` one and expand it to `esc` somewhere upstream. So `esc` is the only "switch to normal mode" command. Just an idea for the future, though.